cfgloader/cfgloader.go

```diff
@@ -105,6 +112,10 @@ func Load(ctx context.Context, cfg any, opt ...Option) error {
 		return fmt.Errorf("failed to load config: %w", err)
 	}
 
+	if d, ok := cfg.(Defaultable); ok {
+		d.SetDefault()
```

Comment:
I'm probably missing the point, but I would have thought that the default would be set first, then overwritten. As I see it here, the defaults come last. This works for things can can be nil-checked like maps and pointers, but for an integer variable, the SetDefault function can't tell whether it's 0 because it was unset or because the environment specifically requested a value of 0.

Reply:
You're totally right.